src/middleware/navigation/has.overseas.entity.middleware.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response, NextFunction } from 'express';
+
+import { logger } from '../../utils/logger';
+import { SECURE_UPDATE_FILTER_URL } from '../../config';
+import { getApplicationData } from "../../utils/application.data";
+import { checkOverseasEntityDetailsEntered, NavigationErrorMessage } from './check.condition';
+
+export const hasOverseasEntity = (req: Request, res: Response, next: NextFunction): void => {
+  try {
+    if ( !checkOverseasEntityDetailsEntered(getApplicationData(req.session)) ) {
+      logger.infoRequest(req, NavigationErrorMessage);
+      return res.redirect(SECURE_UPDATE_FILTER_URL);
```

Comment:
Would it make more sense to set this to the update landing page URL? That way it wont need to be updated if the update landing page is changed.